src/Illuminate/Database/Query/Builder.php

```diff
@@ -1763,13 +1763,14 @@ public function chunkById($count, callable $callback, $column = 'id')
         $lastId = null;
 
         $results = $this->forPageAfterId($count, 0, $column)->get();
+        $columnSuffix = last(explode('.', $column));
```

Comment:
Interesting point, one would certainly expect so. Apparently being wrapped in the [helper function](https://github.com/laravel/framework/blob/be353964453ee0f750e6562419a32d387e8909f0/src/Illuminate/Support/helpers.php#L569-L572) prevents the warning since this creates a new variable.